pkg/controllers/managementuser/rbac/handler_base_test.go

```diff
@@ -0,0 +1,94 @@
+package rbac
+
+import (
+	"fmt"
+	"testing"
+
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	fakes "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3/fakes"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var roles = map[string]*v3.RoleTemplate{
+	"recurse1": {
+		RoleTemplateNames: []string{"recurse2"},
+	},
+	"recurse2": {
+		RoleTemplateNames: []string{"recurse1"},
+	},
+	"non-recurse": {},
+	"inherit non-recurse": {
+		RoleTemplateNames: []string{"non-recurse"},
+	},
+}
+
+func Test_manager_gatherRoles(t *testing.T) {
```

Comment:
I use VSCode to generate the boilerplate for tests. The `manager` is something I haven't seen it do before, but the underscores are because it's a private method and by using camel case you change the name of the function being test, but you also can't have a lowercase after `Test` according to the go docs. If this was a test that only test 1 thing, then I wouldn't have a problem sticking to camel case, but since I'm using a loop and `Run`, I think the name should accuretly reflect which function is being tested. I will remove the manager part though, don't think we need reference to the receiver.